Production build clarification [Epic 2]

## Issue - Bug

Something about the CircleCI production build isn't quite right. Currently on CircleCI, we're performing a production build as:

```
npm run build
```

Then we're checking in the `build/` directory so it may be deployed by dokku to the server.

**❓Question❓:** how does this need to change? Specifically...

...do we need to change the build command to be `npm run build:dll && npm run build`?
...do we need to also make the `dll/` directory available on the server?

- **OS:** Linux

Ah good, I didn't think so.

Something odd is going on, it looks like the deploy config is trying to start the development server in the container:

```
remote:        1488139943506 - Origin: Worker (PID 230)        
remote:           [Error] Error: /app/webpack/webpack.config.dev.js: Cannot find module '../dll/vendors.json' from '/app/webpack'        
remote:            at Function.module.exports [as sync] (/app/node_modules/resolve/lib/sync.js:37:15)        
remote:            at resolveModuleSource (/app/src/server/init.js:13:22)        
remote:            at File.resolveModuleSource (/app/node_modules/babel-core/lib/transformation/file/index.js:333:39)        
remote:            at File.enter (/app/node_modules/babel-core/lib/transformation/file/metadata.js:26:32)        
remote:            at NodePath._call (/app/node_modules/babel-traverse/lib/path/context.js:76:18)        
remote:            at NodePath.call (/app/node_modules/babel-traverse/lib/path/context.js:48:17)        
remote:            at NodePath.visit (/app/node_modules/babel-traverse/lib/path/context.js:105:12)        
remote:            at TraversalContext.visitQueue (/app/node_modules/babel-traverse/lib/context.js:150:16)        
remote:            at TraversalContext.visitMultiple (/app/node_modules/babel-traverse/lib/context.js:103:17)        
remote:            at TraversalContext.visit (/app/node_modules/babel-traverse/lib/context.js:190:19)        
remote:            at Function.traverse.node (/app/node_modules/babel-traverse/lib/index.js:114:17)        
remote:            at traverse (/app/node_modules/babel-traverse/lib/index.js:79:12)        
remote:            at NodePath.traverse (/app/node_modules/babel-traverse/lib/path/index.js:144:25)        
remote:            at File.getMetadata (/app/node_modules/babel-core/lib/transformation/file/index.js:211:17)        
remote:            at File._addAst (/app/node_modules/babel-core/lib/transformation/file/index.js:542:10)        
remote:            at File.addAst (/app/node_modules/babel-core/lib/transformation/file/index.js:547:10)        
remote:            at File.parseCode (/app/node_modules/babel-core/lib/transformation/file/index.js:613:10)        
remote:            at /app/node_modules/babel-core/lib/transformation/pipeline.js:49:12        
remote:            at File.wrap (/app/node_modules/babel-core/lib/transformation/file/index.js:574:16)        
remote:            at Pipeline.transform (/app/node_modules/babel-core/lib/transformation/pipeline.js:47:17)        
remote:            at Object.transformFileSync (/app/node_modules/babel-core/lib/api/node.js:138:10)        
remote:            at compile (/app/node_modules/babel-register/lib/node.js:118:20)        
remote:            at loader (/app/node_modules/babel-register/lib/node.js:144:14)        
remote:            at Object.require.extensions.(anonymous function) [as .js] (/app/node_modules/babel-register/lib/node.js:154:7)        
remote:            at Module.load (module.js:488:32)        
remote:            at tryModuleLoad (module.js:447:12)        
remote:            at Function.Module._load (module.js:439:3)        
remote:            at Module.require (module.js:498:17)        
remote:            at require (internal/module.js:20:19)        
remote:            at Object.<anonymous> (/app/src/server/worker.js:9:1) 
```

🤔

More data on this. Odd, as the production config is being run. Any ideas?

```
remote: =====> web web container output:        
remote:        > parabol-action@0.16.0 start /app        
remote:        > NODE_ENV=production node ./src/server/server.babel.js        
remote:           [Busy] Launching SocketCluster        
remote:           >> Broker PID: 217        
remote:           [Active] SocketCluster started        
remote:                    Version: 5.7.1        
remote:                    Environment: dev        
remote:                    WebSocket engine: uws        
remote:                    Port: 5000        
remote:                    Master PID: 211        
remote:                    Worker count: 1        
remote:                    Broker count: 1        
remote:        1488161161161 - Origin: Worker (PID 229)        
remote:           [Error] Error: /app/webpack/webpack.config.dev.js: Cannot find module '../dll/vendors.json' from '/app/webpack'       
```

--- a/src/server/worker.js
+++ b/src/server/worker.js
@@ -6,7 +6,6 @@ import bodyParser from 'body-parser';
 import jwt from 'express-jwt';
 import favicon from 'serve-favicon';
 import raven from 'raven';
-import config from '../../webpack/webpack.config.dev';
 import createSSR from './createSSR';
 import emailSSR from './emailSSR';
 import {clientSecret as secretKey} from './utils/auth0Helpers';
@@ -35,6 +34,7 @@ export function run(worker) {
 
   // HMR
   if (!PROD) {
+    const config = require('../../webpack/webpack.config.dev').default;
     const compiler = webpack(config);
     /* eslint-disable global-require */
     app.use(require('webpack-dev-middleware')(compiler, {

